synapse/http/proxyagent.py

```diff
@@ -127,6 +140,27 @@ def __init__(
         self._policy_for_https = contextFactory
         self._reactor = reactor
 
+        self._federation_proxy_endpoint: Optional[IStreamClientEndpoint] = None
+        if federation_proxies:
+            endpoints = []
+            for federation_proxy in federation_proxies:
+                endpoint = HostnameEndpoint(
+                    self.proxy_reactor,
+                    federation_proxy.host,
+                    federation_proxy.port,
+                )
+
+                if federation_proxy.tls:
+                    tls_connection_creator = self._policy_for_https.creatorForNetloc(
+                        federation_proxy.host,
```

Comment:
@realtyem mentioned that we probably need to copy what https://github.com/matrix-org/synapse/pull/15746/files did for TLS to work properly. `host` comes in as a `str` and `creatorForNetloc(...)` is expecting `bytes`.


```suggestion
                        federation_proxy.host.encode("utf-8"),
```

Reply:
Fixed in https://github.com/matrix-org/synapse/pull/15886